lib/src/group_changes.dart

```diff
@@ -0,0 +1,307 @@
+            .toList();
+        final configs = <String>[];
+        before = MinMax();
+        final after = MinMax();
+        for (var change in firstSection) {
+          configs.add(change["configuration"]);
+          before.add(hashIndex[change["previous_commit_hash"]]);
+          after.add(hashIndex[change["commit_hash"]]);
+        }
+        configs.sort();
+        final formattedSet = configs.join(",<br>");
+        configSets[formattedSet] = configs;
+        final summaryData =
+            ByConfigSetData(formattedSet, test, results, before, after);
+        byBlamelist
+            .putIfAbsent(summaryData.after.max,
+                () => Map<int, Map<String, List<ByConfigSetData>>>())
+            .putIfAbsent(summaryData.before.min,
+                () => Map<String, List<ByConfigSetData>>())
+            .putIfAbsent(
+                summaryData.formattedConfigSet, () => List<ByConfigSetData>())
+            .add(summaryData);
+      }
+    }
+  }
+  return byBlamelist;
+}
+
+String prelude() => '''
+<!DOCTYPE html><html><head><title>Results Feed</title>
+<style>
+  span.indent   {padding-left: 4em;}
+  textarea      {font-family: monospace;}
+  span.commit   {font-family: monospace; color: gray;}
+  td            {font-family: monospace;}
+  div.blamelist {font-family: monospace; background-color: lightgray; padding: 10px;}
+  h1            {color: blue;}
+  td            {vertical-align: top;}
+  td.outer      {padding: 10px;}
+  td.nopad      {padding: 0px;}
+  td.nomatch    {background-color: Salmon;}
+  td.match      {background-color: PaleGreen;}
+</style>
+<script>function showBlamelist(id) {
+  if (document.getElementById(id + "-off").style.display == "none") {
+    document.getElementById(id + "-on").style.display = "none";
+    document.getElementById(id + "-off").style.display = "block";
+  } else {
+    document.getElementById(id + "-off").style.display = "none";
+    document.getElementById(id + "-on").style.display = "block";
+  }
+}
+
+function showInline(id) {
+  if (document.getElementById(id + "-off").style.display == "none") {
+    document.getElementById(id + "-on").style.display = "none";
+    document.getElementById(id + "-off").style.display = "inline";
+  } else {
+    document.getElementById(id + "-off").style.display = "none";
+    document.getElementById(id + "-on").style.display = "inline";
+  }
+}
+</script>
+</head><body><h1>Results Feed</h1>
+''';
+
+String postlude() => '''</body></html>''';
+
+String htmlPage(Map<int, Map<int, Map<String, List<ByConfigSetData>>>> data,
+    List<String> hashes, List<String> commitData, List<String> reviewLinks) {
+  StringBuffer page = StringBuffer(prelude());
+
+  page.write("<table>");
+  var after = MinMax();
+  data.keys.forEach(after.add);
+  for (var afterKey = 0; afterKey <= after.max; ++afterKey) {
+    // Info from git about this commit:
+    page.write("<span class='commit'><a href='${reviewLinks[afterKey]}'>"
```

Comment:
`<span>` is not not allowed inside `<table>` and the page fails W3 validation.

You should use a proper html escape function.